x-pack/plugins/observability_solution/dataset_quality/public/utils/flatten_stats.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DataStreamType } from '../../common/types';
+
+export function flattenStats<T>(stats: Record<DataStreamType, T[]>): T[] {
```

Comment:
What about this

```js
export function flattenStats<T>(stats: Record<DataStreamType, T[]>): T[] {
  return Object.values(stats)
    .flat()
    .map((dataStream) => ({ ...dataStream, type: dataStream.type }));
}
```

Reply:
1 more thing, we should not return T[] as we are definitely adding `type` to the returned value

Secondly while trying to solve the type issue for the above implementation, i found one more implementation

```js
export function flattenStats<T>(stats: Record<DataStreamType, T[]>): Array<T & { type: DataStreamType }> {
  return Object.entries(stats)
    .flatMap(([type, dataStreams]) =>
      dataStreams.map((dataStream) => ({ ...dataStream, type: type as DataStreamType }))
    );
}
```

Reply:
The first one you propose is missing something, because `T` doesn't have a `type`. The second one is 👌🏼 (:chefs-kiss:)

Reply:
Addressed in [f50a34b](https://github.com/elastic/kibana/pull/190043/commits/f50a34b685f6699fc1cc27be2518ff2bc115fbe0)